app/controllers/gobierto_data/api/v1/datasets_controller.rb

```diff
@@ -148,6 +148,17 @@ def update
           end
         end
 
+        # DELETE /api/v1/data/datasets/:dataset_slug
+        def destroy
+          find_item
```

Comment:
I'd prefer using active record `dependent: :destroy` in the relationship: https://edgeguides.rubyonrails.org/association_basics.html#options-for-belongs-to-dependent

You approach has two issues:

- it's not executed in a single transaction, so if anything fails it might left the database in an inconsistent state
- destroy_all doesn't execute callbacks